pkg/util/wait/wait.go

```diff
@@ -59,16 +75,16 @@ func PollInfinite(interval time.Duration, condition ConditionFunc) error {
 // should be executed and is closed when the last test should be invoked.
 type WaitFunc func() <-chan struct{}
 
-// WaitFor gets a channel from wait(), and then invokes c once for every value
-// placed on the channel and once more when the channel is closed.  If c
-// returns an error the loop ends and that error is returned, and if c returns
+// WaitFor gets a channel from wait(), and then invokes fn once for every value
+// placed on the channel and once more when the channel is closed.  If fn
+// returns an error the loop ends and that error is returned, and if fn returns
 // true the loop ends and nil is returned. ErrWaitTimeout will be returned if
-// the channel is closed without c ever returning true.
```

Comment:
> ErrWaitTimeout will be returned if the channel is closed without c ever returning true.

Drive by nit, this is one of my favorite errors: timed out waiting for the condition.
Can we please `return fmt.Errorf("%v: %v", ErrWaitTimeout, err)`, though that might break anyone checking equality with ErrWaitTimeout.